.github/workflows/build-and-test.yml

```diff
@@ -1,415 +1,540 @@
-                os: [windows-latest, macos-latest]
-
-        steps:
-            - uses: actions/checkout@v2
-              with:
-                  # Fetch complete history for accurate versioning
-                  fetch-depth: 0
-
-              # Taken from https://github.com/actions/cache/blob/master/examples.md#python---pip
-            - name: Restore pip cache
-              uses: actions/cache@v2
-              with:
-                  path: ~\AppData\Local\pip\Cache
-                  key: ${{ runner.os }}-py${{ matrix.python-version }}-${{ matrix.python-architecture }}-pip-${{ hashFiles('**/requirements*.txt') }}-ui-tests
-
-            - name: Setup conda environment
-              uses: conda-incubator/setup-miniconda@v2
-              with:
-                  activate-environment: ${{ matrix.os }}-test-env
-                  auto-update-conda: true
-                  python-version: ${{ matrix.python-version }}
-                  channels: conda-forge
-
-            - name: Install build dependencies
-              shell: bash -l {0}
-              run: |
-                  conda install nomkl requests # make sure numpy is w/out MKL
-                  conda upgrade -y pip setuptools
-
-            - name: Install runtime dependencies
-              shell: bash -l {0}
-              run: |
-                  python ./scripts/convert-requirements-to-conda-yml.py \
-                    requirements.txt requirements-dev.txt \
-                    requirements-gui.txt > requirements-all.yml
-                  conda env update --file requirements-all.yml
-                  make install
-
-            - name: Install Node.js
-              uses: actions/setup-node@v2
-              with:
-                  node-version: 14.x
-
-            - name: Install Workbench Dependencies
-              working-directory: ./workbench
-              run: |
-                  yarn config set network-timeout 600000 -g
-                  yarn install
-
-            - name: run workbench tests
-              shell: bash -l {0}
-              working-directory: ./workbench
-              env:
-                  CI: true
-              run: yarn test
+name: Run Tests and Build
+
+on: [push, pull_request]
+
+defaults:
+  run:
+    shell: bash -l {0}
+
+concurrency:
+  # make sure only one run of this workflow for a given PR or a given branch
+  # can happen at one time. previous queued or started runs will be cancelled.
+  # github.workflow is the workflow name
+  # github.ref is the ref that triggered the workflow run
+  # on push, this is refs/heads/<branch name>
+  # on pull request, this is refs/pull/<pull request number>/merge
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  # build: dependency of make install
+  # nomkl: make sure numpy w/out mkl
+  # setuptools_scm: needed for versioning to work
+  DEFAULT_DEPENDENCIES: build nomkl setuptools_scm
+  PYTHON_VERSION: 3.9

```

Comment:
I think of the difference as that the `PYTHON_VERSION` is the latest supported version, used for jobs that only run on one python version. And the `matrix.python-version` is one of `[3.7, 3.8, 3.9]`, used for jobs that have to run multiple python versions.

I renamed `PYTHON_VERSION` to `LATEST_SUPPORTED_PYTHON_VERSION`. And added a new variable `SUPPORTED_PYTHON_VERSIONS: [3.7, 3.8, 3.9]`. Those values then becomes the `matrix.python-version` within a job. Does that make more sense?

Side note, should we make the supported python versions official with the setuptools [`python_requires`](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#python-requirement) config value?

Maybe eventually we could check in an `environment.yml` alongside our `requirements.txt`s. Since that can handle specifying our supported python versions and the gdal binaries.

Reply:
Also, is it time to add 3.10 and drop 3.7?

Reply:
> Side note, should we make the supported python versions official with the setuptools [python_requires](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#python-requirement) config value?

Yeah, this would be wise to add since we use f-strings in a lot of places (>= python3.6).

> I renamed PYTHON_VERSION to LATEST_SUPPORTED_PYTHON_VERSION ... Does that make more sense?

Yep, thanks!

> Also, is it time to add 3.10 and drop 3.7?

Feel free to add it to the build matrix if you like!  I see we also have https://github.com/natcap/invest/issues/672 for testing against 3.10, and there do look to be a few other changes needed that are beyond the scope of this PR and should probably be made independently of this refactor (like updates to the pyinstaller .spec file).

For 3.7, I personally might prefer to leave it in until we have a reason to drop it (like some super-useful syntax change that's only available in 3.8+).  3.7 is still actively supported by the python project, so it seems reasonable to keep support for now so long as it isn't a headache.

Reply:
Oh, I forgot #672 wasn't done yet